lib/ansible/modules/web_infrastructure/jenkins_node.py

```diff
@@ -0,0 +1,370 @@
+    type: bool
+maintenance:
+    description: Indicate status of maintenance flag.
+    sample: true`
+    returned: success
+    type: bool
+label:
+    description: Name of maintenance label.
+    type: str
+    returned: success
+    sample: maintenance
+changed:
+    description: indicates if task ran to change the state.
+    returned: success
+    sample: true
+    type: bool
+'''
+
+import time
+import re
+import traceback
+try:
+    import jenkins
+    python_jenkins_installed = True
+except ImportError:
+    JENKINS_IMP_ERR = traceback.format_exc()
+    python_jenkins_installed = False
+
+import xml.etree.ElementTree as ET
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+
+
+def test_dependencies(module):
+    # Test if python-jenkins python package is present.
+    if not python_jenkins_installed:
+        module.fail_json(
+            msg=missing_required_lib("python-jenkins",
+                                     url="https://python-jenkins.readthedocs.io/en/latest/install.html"),
+            exception=JENKINS_IMP_ERR)
+
+
+def host_config(module, jenkins_server, jenkins_username, jenkins_token):
+    # Create jenkins server object.
+    try:
+        server = jenkins.Jenkins(jenkins_server,
+                                 username=jenkins_username,
+                                 password=jenkins_token)
+    except Exception as e:
+        module.fail_json(msg="Unable to contact jenkins server: {0}".format(e))
+    return server
+
+
+def jenkins_node_info(host, server_node, module):
+    # Get jenkins nodes information.
+    try:
+        node_info = server_node.get_node_info(host)
+    except Exception as e:
+        module.fail_json(msg="Unable to contact jenkins node {0}: {1},".format(host, e))
+    return node_info
+
+
+def delete_node(server_node, host, module):
+    # Delete jenkins node from jenkins server.
+    try:
+        server_node.delete_node(host)
+    except jenkins.JenkinsException:
+        module.fail_json(msg="Could not delete {0}'.format(host)".format(host))
+
+
+def set_node_status(server_node, host, dict_online):
+    # Disconnect jenkins node with message from jenkins server.
+    if dict_online['state'] in ['no', False, 'false']:
+        server_node.disable_node(host, dict_online['msg'])
+    else:
+        server_node.enable_node(host)
+
+
+def node_config_change(host, server_node, dict_label,
+                       result, module):
+    try:
+        tree = ET.fromstring(server_node.get_node_config(host))
+        root = ET.ElementTree(element=tree).getroot()
+    except Exception as e:
+        module.fail_json(msg="Unable to contact jenkins node {0}: (get_node_config): {1}".format(host, e))
+    try:
+        for child in root:
+            if child.tag == 'label':
+                current_label = child.text
+                # list with unique elements and not extra spaces.
+                current_label_list = list(set(list(filter(None, current_label.split(" ")))))
```

Comment:
Filter None is used to remove "None". Sometimes a Jenkins node is not assigned a label (This is common) or worse its been misconfigured to return None (This also happens). This can be seen when Jenkins would return "null" from groovy. Since we use Python we get "None". I am simply trying to remove "None".